2005-06-08  Dmitry V. Levin  <ldv@altlinux.org>

	Remove TF flag from those syscalls which have no filename argument.
	* freebsd/i386/syscallent.h: Remove TF flag from fstat, pread,
	pwrite, fstat, sendfile.
	* linux/alpha/syscallent.h: Remove TF flag from read, write,
	pread, pwrite, sendfile.
	* linux/hppa/syscallent.h: Remove TF flag from read, write,
	pread, pwrite, fstat64, sendfile, ftruncate64.
	* linux/ia64/syscallent.h: Remove TF flag from read, write,
	sendfile, fstat, fadvise64.
	* linux/mips/syscallent.h: Remove TF flag from read, write,
	fstatfs, fstat, pread, pwrite, sendfile, ftruncate64, fstat64,
	sendfile64, fadvise64, fstatfs64.
	* linux/powerpc/syscallent.h: Remove TF flag from read, write,
	fstat, pread, pwrite, sendfile, ftruncate64, fstat64, sendfile64,
	fadvise64, fstatfs64, fadvise64_64.
	* linux/s390/syscallent.h: Remove TF flag from pread, pwrite,
	sendfile, ftruncate64, fstat64, sendfile64.
	* linux/s390x/syscallent.h: Remove TF flag from pread, pwrite,
	sendfile, sendfile64.
	* linux/sh/syscallent.h: Remove TF flag from pread, pwrite,
	sendfile, fstat64.
	* linux/sh64/syscallent.h: Remove TF flag from pread, pwrite,
	sendfile, ftruncate64, fstat64.
	* linux/sparc/syscallent.h: Remove TF flag from sendfile, fstat64,
	pread, pwrite, sendfile64.
	* linux/sparc64/syscallent2.h: Likewise.
	* linux/sparc64/syscallent.h: Likewise.
	* linux/syscallent.h: Remove TF flag from pread, pwrite, sendfile,
	ftruncate64, fstat64, sendfile64, fadvise64, fadvise64, fstatfs64,
	fadvise64_64.
	* linux/x86_64/syscallent.h: Remove TF flag from pread, pwrite,
	sendfile, fadvise64_64.
	* svr4/syscallent.h: Remove TF flag from pread, pwrite, ftruncate,
	fstatvfs64, ftruncate64.
	Fixes RH#159340.

diff -upk.orig strace-4.5.12.orig/freebsd/i386/syscallent.h strace-4.5.12/freebsd/i386/syscallent.h
--- strace-4.5.12.orig/freebsd/i386/syscallent.h	2001-03-08 17:27:20 +0000
+++ strace-4.5.12/freebsd/i386/syscallent.h	2005-06-08 08:55:32 +0000
@@ -64,7 +64,7 @@
   { 3,	TF|TP,	sys_execve,	"execve"	}, /* 59 */
   { 1,	0,	sys_umask,	"umask"	}, /* 60 */
   { 1,	TF,	sys_chroot,	"chroot"	}, /* 61 */
-  { 2,	TF,	sys_fstat,	"fstat?"	}, /* 62 */
+  { 2,	0,	sys_fstat,	"fstat?"	}, /* 62 */
   { 4,	0,	sys_getkerninfo,	"getkerninfo?"	}, /* 63 */
   { 1,	0,	sys_getpagesize,	"getpagesize?"	}, /* 64 */
   { 3,	0,	sys_msync,	"msync"	}, /* 65 */
@@ -175,8 +175,8 @@
   { 6,	TI,	sys_msgsys,	"msgsys"	}, /* 170 */
   { 4,	TI,	sys_shmsys,	"shmsys"	}, /* 171 */
   { -1,	0,	printargs,	"SYS_172"	}, /* 172 */
-  { 5,	TF,	sys_pread,	"pread"	}, /* 173 */
-  { 5,	TF,	sys_pwrite,	"pwrite"	}, /* 174 */
+  { 5,	0,	sys_pread,	"pread"	}, /* 173 */
+  { 5,	0,	sys_pwrite,	"pwrite"	}, /* 174 */
   { -1,	0,	printargs,	"SYS_175"	}, /* 175 */
   { 1,	0,	sys_ntp_adjtime,	"ntp_adjtime"	}, /* 176 */
   { -1,	0,	printargs,	"SYS_177"	}, /* 177 */
@@ -191,7 +191,7 @@
   { -1,	0,	printargs,	"SYS_186"	}, /* 186 */
   { -1,	0,	printargs,	"SYS_187"	}, /* 187 */
   { 2,	TF,	sys_stat,	"stat"	}, /* 188 */
-  { 2,	TF,	sys_fstat,	"fstat"	}, /* 189 */
+  { 2,	0,	sys_fstat,	"fstat"	}, /* 189 */
   { 2,	TF,	sys_lstat,	"lstat"	}, /* 190 */
   { 2,	TF,	sys_pathconf,	"pathconf"	}, /* 191 */
   { 2,	0,	sys_fpathconf,	"fpathconf"	}, /* 192 */
@@ -338,7 +338,7 @@
   { 1,	0,	sys_sched_get_priority_min,	"sched_get_priority_min"	}, /* 333 */
   { 2,	0,	sys_sched_rr_get_interval,	"sched_rr_get_interval"	}, /* 334 */
   { 2,	0,	sys_utrace,	"utrace"	}, /* 335 */
-  { 7,	TF,	sys_sendfile,	"sendfile"	}, /* 336 */
+  { 7,	0,	sys_sendfile,	"sendfile"	}, /* 336 */
   { 3,	0,	sys_kldsym,	"kldsym"	}, /* 337 */
   { 1,	0,	sys_jail,	"jail"	}, /* 338 */
   { -1,	0,	printargs,	"SYS_339"	}, /* 339 */
diff -upk.orig strace-4.5.12.orig/linux/alpha/syscallent.h strace-4.5.12/linux/alpha/syscallent.h
--- strace-4.5.12.orig/linux/alpha/syscallent.h	2005-05-09 08:02:05 +0000
+++ strace-4.5.12/linux/alpha/syscallent.h	2005-06-10 11:37:30 +0000
@@ -31,8 +31,8 @@
 	{ 6,	0,	printargs,		"osf_syscall"		}, /* 0, not implemented */
 	{ 1,	TP,	sys_exit,		"exit"			}, /* 1 */
 	{ 0,	TP,	sys_fork,		"fork"			}, /* 2 */
-	{ 3,	TF,	sys_read,		"read"			}, /* 3 */
-	{ 3,	TF,	sys_write,		"write"			}, /* 4 */
+	{ 3,	0,	sys_read,		"read"			}, /* 3 */
+	{ 3,	0,	sys_write,		"write"			}, /* 4 */
 	{ 5,	0,	printargs,		"osf_old_open"		}, /* 5, not implemented */
 	{ 1,	0,	sys_close,		"close"			}, /* 6 */
 	{ 4,	TP,	sys_osf_wait4,		"osf_wait4"		}, /* 7 */
@@ -377,8 +377,8 @@
 	{ 5,	0,	printargs,		"pciconfig_write"	}, /* 346 */
 	{ 5,	0,	sys_query_module,	"query_module"		}, /* 347 */
 	{ 5,	0,	printargs,		"prctl"			}, /* 348 */
-	{ 5,	TF,	sys_pread,		"pread"			}, /* 349 */
-	{ 5,	TF,	sys_pwrite,		"pwrite"		}, /* 350 */
+	{ 5,	0,	sys_pread,		"pread"			}, /* 349 */
+	{ 5,	0,	sys_pwrite,		"pwrite"		}, /* 350 */
 	{ 1,	TS,	printargs,		"rt_sigreturn"		}, /* 351 */
 	{ 4,	TS,	sys_rt_sigaction,	"rt_sigaction"		}, /* 352 */
 	{ 4,	TS,	sys_rt_sigprocmask,	"rt_sigprocmask"	}, /* 353 */
@@ -398,7 +398,7 @@
 	{ 2,	0,	sys_getcwd,		"getcwd"		}, /* 367 */
 	{ 2,	0,	sys_capget,		"capget"		}, /* 368 */
 	{ 2,	0,	sys_capset,		"capset"		}, /* 369 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"		}, /* 370 */
+	{ 4,	0,	sys_sendfile,		"sendfile"		}, /* 370 */
 	{ 3,	0,	printargs,		"setresgid"		}, /* 371 */
 	{ 3,	0,	printargs,		"getresgid"		}, /* 372 */
 	{ 4,	0,	printargs,		"dipc"			}, /* 373, not implemented */
diff -upk.orig strace-4.5.12.orig/linux/hppa/syscallent.h strace-4.5.12/linux/hppa/syscallent.h
--- strace-4.5.12.orig/linux/hppa/syscallent.h	2005-05-09 08:02:04 +0000
+++ strace-4.5.12/linux/hppa/syscallent.h	2005-06-10 11:37:30 +0000
@@ -8,8 +8,8 @@
 	{ 5,	0,	printargs,		"SYS_0"			}, /* 0 */
 	{ 1,	TP,	sys_exit,		"exit"			}, /* 1 */
 	{ 0,	TP,	sys_fork,		"fork"			}, /* 2 */
-	{ 3,	TF,	sys_read,		"read"			}, /* 3 */
-	{ 3,	TF,	sys_write,		"write"			}, /* 4 */
+	{ 3,	0,	sys_read,		"read"			}, /* 3 */
+	{ 3,	0,	sys_write,		"write"			}, /* 4 */
 	{ 3,	TF,	sys_open,		"open"			}, /* 5 */
 	{ 1,	0,	sys_close,		"close"			}, /* 6 */
 	{ 3,	TP,	sys_waitpid,		"waitpid"		}, /* 7 */
@@ -113,11 +113,11 @@
 	{ 2,	0,	sys_getitimer,		"getitimer"		}, /* 105 */
 	{ 2,	0,	sys_capget,		"capget"		}, /* 106 */
 	{ 2,	0,	sys_capset,		"capset"		}, /* 107 */
-	{ 5,	TF,	sys_pread,		"pread"			}, /* 108 */
-	{ 5,	TF,	sys_pwrite,		"pwrite"		}, /* 109 */
+	{ 5,	0,	sys_pread,		"pread"			}, /* 108 */
+	{ 5,	0,	sys_pwrite,		"pwrite"		}, /* 109 */
 	{ 2,	0,	sys_getcwd,		"getcwd"		}, /* 110 */
 	{ 0,	0,	sys_vhangup,		"vhangup"		}, /* 111 */
-	{ 2,	TF,	sys_fstat64,		"fstat64"		}, /* 112 */
+	{ 2,	0,	sys_fstat64,		"fstat64"		}, /* 112 */
 	{ 0,	0,	sys_vfork,		"vfork"			}, /* 113 */
 	{ 4,	TP,	sys_wait4,		"wait4"			}, /* 114 */
 	{ 1,	0,	sys_swapoff,		"swapoff"		}, /* 115 */
@@ -127,7 +127,7 @@
 	{ 3,	0,	sys_madvise,		"madvise"		}, /* 119 */
 	{ 5,	TP,	sys_clone,		"clone"			}, /* 120 */
 	{ 2,	0,	sys_setdomainname,	"setdomainname"		}, /* 121 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"		}, /* 122 */
+	{ 4,	0,	sys_sendfile,		"sendfile"		}, /* 122 */
 	{ 6,	TN,	sys_recvfrom,		"recvfrom"		}, /* 123 */
 	{ 1,	0,	sys_adjtimex,		"adjtimex"		}, /* 124 */
 	{ 3,	0,	sys_mprotect,		"mprotect"		}, /* 125 */
@@ -205,7 +205,7 @@
 	{ 5,	0,	sys_putpmsg,		"putpmsg"		}, /* 197 */
 	{ 2,	TF,	sys_lstat64,		"lstat64"		}, /* 198 */
 	{ 3,	TF,	sys_truncate64,		"truncate64"		}, /* 199 */
-	{ 3,	TF,	sys_ftruncate64,	"ftruncate64"		}, /* 200 */
+	{ 3,	0,	sys_ftruncate64,	"ftruncate64"		}, /* 200 */
 	{ 3,	0,	sys_getdents64,		"getdents64"		}, /* 201 */
 	{ 3,	0,	sys_fcntl,		"fcntl64"		}, /* 202 */
 	{ 5,	0,	printargs,		"attrctl"		}, /* 203 */
@@ -214,7 +214,7 @@
 	{ 0,	0,	printargs,		"gettid"		}, /* 206 */
 	{ 4,	0,	sys_readahead,		"readahead"		}, /* 207 */
 	{ 2,	TS,	sys_kill,		"tkill"			}, /* 208 */
-	{ 4,	TF,	sys_sendfile,		"sendfile64"		}, /* 209 */
+	{ 4,	0,	sys_sendfile,		"sendfile64"		}, /* 209 */
 	{ 5,	0,	sys_futex,		"futex"			}, /* 210 */
 	{ 3,	0,	sys_sched_setaffinity,	"sched_setaffinity"	}, /* 211 */
 	{ 3,	0,	sys_sched_getaffinity,	"sched_getaffinity"	}, /* 212 */
diff -upk.orig strace-4.5.12.orig/linux/ia64/syscallent.h strace-4.5.12/linux/ia64/syscallent.h
--- strace-4.5.12.orig/linux/ia64/syscallent.h	2005-05-09 08:02:05 +0000
+++ strace-4.5.12/linux/ia64/syscallent.h	2005-06-10 11:37:30 +0000
@@ -929,8 +929,8 @@
 	{ 8,	0,	printargs,		"SYS_1023"	}, /* 1023 */
 	{ 0,	0,	printargs,		"ni_syscall"	}, /* 1024 */
 	{ 1,	TP,	sys_exit,		"exit"		}, /* 1025 */
-	{ 3,	TF,	sys_read,		"read"		}, /* 1026 */
-	{ 3,	TF,	sys_write,		"write"		}, /* 1027 */
+	{ 3,	0,	sys_read,		"read"		}, /* 1026 */
+	{ 3,	0,	sys_write,		"write"		}, /* 1027 */
 	{ 3,	TF,	sys_open,		"open"		}, /* 1028 */
 	{ 1,	0,	sys_close,		"close"		}, /* 1029 */
 	{ 2,	0,	sys_creat,		"creat"		}, /* 1030 */
@@ -1090,7 +1090,7 @@
 	{ 2,	TF,	sys_getcwd,		"getcwd"	}, /* 1184 */
 	{ 2,	0,	sys_capget,		"capget"	}, /* 1185 */
 	{ 2,	0,	sys_capset,		"capset"	}, /* 1186 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"	}, /* 1187 */
+	{ 4,	0,	sys_sendfile,		"sendfile"	}, /* 1187 */
 	{ 5,	TN,	printargs,		"getpmsg"	}, /* 1188 */
 	{ 5,	TN,	printargs,		"putpmsg"	}, /* 1189 */
 	{ 3,	TN,	sys_socket,		"socket"	}, /* 1190 */
@@ -1115,7 +1115,7 @@
 	{ 3,	0,	sys_madvise,		"madvise"	}, /* 1209 */
 	{ 2,	TF,	sys_stat,		"stat"		}, /* 1210 */
 	{ 2,	0,	sys_lstat,		"lstat"		}, /* 1211 */
-	{ 2,	TF,	sys_fstat,		"fstat"		}, /* 1212 */
+	{ 2,	0,	sys_fstat,		"fstat"		}, /* 1212 */
 	{ 6,	TP,	sys_clone,		"clone2"	}, /* 1213 */
 	{ 3,	0,	printargs,		"getdents64"	}, /* 1214 */
 	{ 2,	0,	printargs,		"getunwind"	}, /* 1215 */
@@ -1137,7 +1137,7 @@
 	{ 3,	0,	sys_sched_setaffinity,	"sched_setaffinity"},/* 1231 */
 	{ 3,	0,	sys_sched_getaffinity,	"sched_getaffinity"},/* 1232 */
 	{ 1,	0,	printargs,		"set_tid_address"}, /* 1233 */
-	{ 5,	TF,	sys_fadvise64,		"fadvise64"	}, /* 1234 */
+	{ 5,	0,	sys_fadvise64,		"fadvise64"	}, /* 1234 */
 	{ 3,	TS,	sys_tgkill,		"tgkill"	}, /* 1235 */
 	{ 1,	TP,	sys_exit,		"exit_group"	}, /* 1236 */
 	{ 4,	0,	printargs,		"lookup_dcookie"}, /* 1237 */
diff -upk.orig strace-4.5.12.orig/linux/mips/syscallent.h strace-4.5.12/linux/mips/syscallent.h
--- strace-4.5.12.orig/linux/mips/syscallent.h	2005-05-09 07:54:30 +0000
+++ strace-4.5.12/linux/mips/syscallent.h	2005-06-10 11:37:30 +0000
@@ -4001,8 +4001,8 @@
 	{ 8,	0,	printargs,		"syscall"	}, /* 4000 */ /* start of Linux o32 */
 	{ 1,	TP,	sys_exit,		"exit"		}, /* 4001 */
 	{ 0,	TP,	sys_fork,		"fork"		}, /* 4002 */
-	{ 3,	TF,	sys_read,		"read"		}, /* 4003 */
-	{ 3,	TF,	sys_write,		"write"		}, /* 4004 */
+	{ 3,	0,	sys_read,		"read"		}, /* 4003 */
+	{ 3,	0,	sys_write,		"write"		}, /* 4004 */
 	{ 3,	TF,	sys_open,		"open"		}, /* 4005 */
 	{ 1,	0,	sys_close,		"close"		}, /* 4006 */
 	{ 3,	TP,	sys_waitpid,		"waitpid"	}, /* 4007 */
@@ -4098,7 +4098,7 @@
 	{ 3,	0,	sys_setpriority,	"setpriority"	}, /* 4097 */
 	{ 0,	0,	sys_profil,		"profil"	}, /* 4098 */
 	{ 3,	TF,	sys_statfs,		"statfs"	}, /* 4099 */
-	{ 3,	TF,	sys_fstatfs,		"fstatfs"	}, /* 4100 */
+	{ 3,	0,	sys_fstatfs,		"fstatfs"	}, /* 4100 */
 	{ 0,	0,	sys_ioperm,		"ioperm"	}, /* 4101 */
 	{ 2,	0,	sys_socketcall,		"socketcall"	}, /* 4102 */
 	{ 3,	0,	sys_syslog,		"syslog"	}, /* 4103 */
@@ -4106,7 +4106,7 @@
 	{ 2,	0,	sys_getitimer,		"getitimer"	}, /* 4105 */
 	{ 2,	TF,	sys_stat,		"stat"		}, /* 4106 */
 	{ 2,	TF,	sys_lstat,		"lstat"		}, /* 4107 */
-	{ 2,	TF,	sys_fstat,		"fstat"		}, /* 4108 */
+	{ 2,	0,	sys_fstat,		"fstat"		}, /* 4108 */
 	{ 1,	0,	sys_olduname,		"olduname"	}, /* 4109 */
 	{ 0,	0,	sys_iopl,		"iopl"		}, /* 4110 */
 	{ 0,	0,	sys_vhangup,		"vhangup"	}, /* 4111 */
@@ -4198,22 +4198,22 @@
 	{ 4,	TS,	sys_rt_sigtimedwait,	"rt_sigtimedwait"},/* 4197 */
 	{ 3,	TS,	sys_rt_sigqueueinfo,	"rt_sigqueueinfo"},/* 4198 */
 	{ 2,	TS,	sys_rt_sigsuspend,	"rt_sigsuspend"	}, /* 4199 */
-	{ 6,	TF,	sys_pread,		"pread"		}, /* 4200 */
-	{ 6,	TF,	sys_pwrite,		"pwrite"	}, /* 4201 */
+	{ 6,	0,	sys_pread,		"pread"		}, /* 4200 */
+	{ 6,	0,	sys_pwrite,		"pwrite"	}, /* 4201 */
 	{ 3,	TF,	sys_chown,		"chown"		}, /* 4202 */
 	{ 2,	TF,	sys_getcwd,		"getcwd"	}, /* 4203 */
 	{ 2,	0,	sys_capget,		"capget"	}, /* 4204 */
 	{ 2,	0,	sys_capset,		"capset"	}, /* 4205 */
 	{ 2,	TS,	sys_sigaltstack,	"sigaltstatck"	}, /* 4206 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"	}, /* 4207 */
+	{ 4,	0,	sys_sendfile,		"sendfile"	}, /* 4207 */
 	{ 0,	0,	printargs,		"SYS_4208"	}, /* 4208 */
 	{ 0,	0,	printargs,		"SYS_4209"	}, /* 4209 */
 	{ 6,	0,	sys_mmap,		"mmap"		}, /* 4210 */
 	{ 4,	TF,	sys_truncate64,		"truncate64"	}, /* 4211 */
-	{ 4,	TF,	sys_ftruncate64,	"ftruncate64"	}, /* 4212 */
+	{ 4,	0,	sys_ftruncate64,	"ftruncate64"	}, /* 4212 */
 	{ 2,	TF,	printargs,		"stat64"	}, /* 4213 */
 	{ 2,	TF,	sys_lstat64,		"lstat64"	}, /* 4214 */
-	{ 2,	TF,	sys_fstat64,		"fstat64"	}, /* 4215 */
+	{ 2,	0,	sys_fstat64,		"fstat64"	}, /* 4215 */
 	{ 2,	TF,	sys_pivotroot,		"pivot_root"	}, /* 4216 */
 	{ 3,	0,	printargs,		"mincore"	}, /* 4217 */
 	{ 3,	0,	sys_madvise,		"madvise"	}, /* 4218 */
@@ -4235,7 +4235,7 @@
 	{ 2,	TF,	sys_removexattr,	"lremovexattr"	}, /* 4234 */
 	{ 2,	0,	sys_fremovexattr,	"fremovexattr"	}, /* 4235 */
 	{ 2,	TS,	sys_kill,		"tkill"		}, /* 4236 */
-	{ 5,	TF,	sys_sendfile64,		"sendfile64"	}, /* 4237 */
+	{ 5,	0,	sys_sendfile64,		"sendfile64"	}, /* 4237 */
 	{ 6,	0,	sys_futex,		"futex"		}, /* 4238 */
 	{ 3,	0,	sys_sched_setaffinity,	"sched_setaffinity"}, /* 4239 */
 	{ 3,	0,	sys_sched_getaffinity,	"sched_getaffinity"}, /* 4240 */
@@ -4252,9 +4252,9 @@
 	{ 5,	0,	sys_remap_file_pages,	"remap_file_pages"}, /* 4251 */
 	{ 1,	0,	printargs,		"set_tid_address"}, /* 4252 */
 	{ 0,	0,	printargs,		"restart_syscall"}, /* 4253 */
-	{ 7,	TF,	sys_fadvise64_64,	"fadvise64_64"	}, /* 4254 */
+	{ 7,	0,	sys_fadvise64_64,	"fadvise64_64"	}, /* 4254 */
 	{ 3,	TF,	sys_statfs64,		"statfs64"	}, /* 4255 */
-	{ 2,	TF,	sys_fstatfs64,		"fstatfs64"	}, /* 4256 */
+	{ 2,	0,	sys_fstatfs64,		"fstatfs64"	}, /* 4256 */
 	{ 3,	0,	sys_timer_create,	"timer_create"	}, /* 4257 */
 	{ 4,	0,	sys_timer_settime,	"timer_settime"	}, /* 4258 */
 	{ 2,	0,	sys_timer_gettime,	"timer_gettime"	}, /* 4259 */
diff -upk.orig strace-4.5.12.orig/linux/powerpc/syscallent.h strace-4.5.12/linux/powerpc/syscallent.h
--- strace-4.5.12.orig/linux/powerpc/syscallent.h	2005-05-09 08:02:05 +0000
+++ strace-4.5.12/linux/powerpc/syscallent.h	2005-06-10 11:37:30 +0000
@@ -31,8 +31,8 @@
 	{ 5,	0,	printargs,		"SYS_0"			}, /* 0 */
 	{ 1,	TP,	sys_exit,		"exit"			}, /* 1 */
 	{ 0,	TP,	sys_fork,		"fork"			}, /* 2 */
-	{ 3,	TF,	sys_read,		"read"			}, /* 3 */
-	{ 3,	TF,	sys_write,		"write"			}, /* 4 */
+	{ 3,	0,	sys_read,		"read"			}, /* 3 */
+	{ 3,	0,	sys_write,		"write"			}, /* 4 */
 	{ 3,	TF,	sys_open,		"open"			}, /* 5 */
 	{ 1,	0,	sys_close,		"close"			}, /* 6 */
 	{ 3,	TP,	sys_waitpid,		"waitpid"		}, /* 7 */
@@ -136,7 +136,7 @@
 	{ 2,	0,	sys_getitimer,		"getitimer"		}, /* 105 */
 	{ 2,	TF,	sys_stat,		"stat"			}, /* 106 */
 	{ 2,	TF,	sys_lstat,		"lstat"			}, /* 107 */
-	{ 2,	TF,	sys_fstat,		"fstat"			}, /* 108 */
+	{ 2,	0,	sys_fstat,		"fstat"			}, /* 108 */
 	{ 1,	0,	sys_olduname,		"olduname"		}, /* 109 */
 	{ 5,	0,	printargs,		"SYS_110"		}, /* 110 */
 	{ 0,	0,	sys_vhangup,		"vhangup"		}, /* 111 */
@@ -207,14 +207,14 @@
 	{ 4,	TS,	sys_rt_sigtimedwait,	"rt_sigtimedwait"	}, /* 176 */
 	{ 3,	TS,	sys_rt_sigqueueinfo,	"rt_sigqueueinfo"	}, /* 177 */
 	{ 2,	TS,	sys_rt_sigsuspend,	"rt_sigsuspend"		}, /* 178 */
-	{ 6,	TF,	sys_pread,		"pread64"		}, /* 179 */
-	{ 6,	TF,	sys_pwrite,		"pwrite64"		}, /* 180 */
+	{ 6,	0,	sys_pread,		"pread64"		}, /* 179 */
+	{ 6,	0,	sys_pwrite,		"pwrite64"		}, /* 180 */
 	{ 3,	TF,	sys_chown,		"chown"			}, /* 181 */
 	{ 2,	TF,	sys_getcwd,		"getcwd"		}, /* 182 */
 	{ 2,	0,	sys_capget,		"capget"		}, /* 183 */
 	{ 2,	0,	sys_capset,		"capset"		}, /* 184 */
 	{ 2,	TS,	sys_sigaltstack,	"sigaltstack"		}, /* 185 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"		}, /* 186 */
+	{ 4,	0,	sys_sendfile,		"sendfile"		}, /* 186 */
 	{ 5,	0,	sys_getpmsg,		"getpmsg"		}, /* 187 */
 	{ 5,	0,	sys_putpmsg,		"putpmsg"		}, /* 188 */
 	{ 0,	TP,	sys_vfork,		"vfork"			}, /* 189 */
@@ -222,11 +222,11 @@
 	{ 4,	0,	sys_readahead,		"readahead"		}, /* 190 */
 	{ 6,	0,	sys_mmap,		"mmap2"			}, /* 192 */
 	{ 4,	TF,	sys_truncate64,		"truncate64"		}, /* 193 */
-	{ 4,	TF,	sys_ftruncate64,	"ftruncate64"		}, /* 194 */
+	{ 4,	0,	sys_ftruncate64,	"ftruncate64"		}, /* 194 */
 
 	{ 2,	TF,	sys_stat64,		"stat64"		}, /* 195 */
 	{ 2,	TF,	sys_lstat64,		"lstat64"		}, /* 196 */
-	{ 2,	TF,	sys_fstat64,		"fstat64"		}, /* 197 */
+	{ 2,	0,	sys_fstat64,		"fstat64"		}, /* 197 */
 	{ 5,	0,	printargs,		"pciconfig_read"	}, /* 198 */
 	{ 5,	0,	printargs,		"pciconfig_write"	}, /* 199 */
 	{ 3,	0,	printargs,		"pciconfig_iobase"	}, /* 200 */
@@ -255,14 +255,14 @@
 	{ 3,	0,	sys_sched_getaffinity,	"sched_getaffinity"	}, /* 223 */
 	{ 5,	0,	printargs,		"SYS_224"		}, /* 224 */
 	{ 5,	0,	printargs,		"tux"			}, /* 225 */
-	{ 4,	TF,	sys_sendfile64,		"sendfile64"		}, /* 226 */
+	{ 4,	0,	sys_sendfile64,		"sendfile64"		}, /* 226 */
 	{ 2,	0,	sys_io_setup,		"io_setup"		}, /* 227 */
 	{ 1,	0,	sys_io_destroy,		"io_destroy"		}, /* 228 */
 	{ 5,	0,	sys_io_getevents,		"io_getevents"		}, /* 229 */
 	{ 3,	0,	sys_io_submit,		"io_submit"		}, /* 230 */
 	{ 3,	0,	sys_io_cancel,		"io_cancel"		}, /* 231 */
 	{ 1,	0,	printargs,		"set_tid_address"	}, /* 232 */
-	{ 6,	TF,	sys_fadvise64,		"fadvise64"		}, /* 233 */
+	{ 6,	0,	sys_fadvise64,		"fadvise64"		}, /* 233 */
 	{ 1,	TP,	sys_exit,		"exit_group"		}, /* 234 */
 	{ 4,	0,	printargs,		"lookup_dcookie"	}, /* 235 */
 	{ 1,	0,	sys_epoll_create,	"epoll_create"		}, /* 236 */
@@ -282,8 +282,8 @@
 	{ 3,	TS,	sys_tgkill,		"tgkill"		}, /* 250 */
 	{ 2,	TF,	sys_utimes,		"utimes"		}, /* 251 */
 	{ 3,	TF,	sys_statfs64,		"statfs64"		}, /* 252 */
-	{ 3,	TF,	sys_fstatfs64,		"fstatfs64"		}, /* 253 */
-	{ 6,	TF,	sys_fadvise64_64,	"fadvise64_64"		}, /* 254 */
+	{ 3,	0,	sys_fstatfs64,		"fstatfs64"		}, /* 253 */
+	{ 6,	0,	sys_fadvise64_64,	"fadvise64_64"		}, /* 254 */
 	{ 1,	0,	printargs,		"rtas"			}, /* 255 */
 	{ 5,	0,	printargs,		"debug_setcontext"	}, /* 256 */
 	{ 5,	0,	printargs,		"vserver"		}, /* 257 */
diff -upk.orig strace-4.5.12.orig/linux/s390/syscallent.h strace-4.5.12/linux/s390/syscallent.h
--- strace-4.5.12.orig/linux/s390/syscallent.h	2005-05-09 07:54:30 +0000
+++ strace-4.5.12/linux/s390/syscallent.h	2005-06-10 11:37:30 +0000
@@ -208,24 +208,24 @@
 	{ 4,	TS,	sys_rt_sigtimedwait,	"rt_sigtimedwait"}, /* 177 */
 	{ 3,	TS,	sys_rt_sigqueueinfo,	"rt_sigqueueinfo"}, /* 178 */
 	{ 2,	TS,	sys_rt_sigsuspend,	"rt_sigsuspend"	}, /* 179 */
-	{ 5,	TF,	sys_pread,		"pread"		}, /* 180 */
-	{ 5,	TF,	sys_pwrite,		"pwrite"	}, /* 181 */
+	{ 5,	0,	sys_pread,		"pread"		}, /* 180 */
+	{ 5,	0,	sys_pwrite,		"pwrite"	}, /* 181 */
 	{ 3,	TF,	sys_chown,		"lchown"	}, /* 182 */
 	{ 2,	TF,	sys_getcwd,		"getcwd"	}, /* 183 */
 	{ 2,	0,	sys_capget,		"capget"	}, /* 184 */
 	{ 2,	0,	sys_capset,		"capset"	}, /* 185 */
 	{ 2,	TS,	sys_sigaltstack,	"sigaltstack"	}, /* 186 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"	}, /* 187 */
+	{ 4,	0,	sys_sendfile,		"sendfile"	}, /* 187 */
 	{ 5,	0,	sys_getpmsg,		"getpmsg"	}, /* 188 */
 	{ 5,	0,	sys_putpmsg,		"putpmsg"	}, /* 189 */
 	{ 0,	TP,	sys_vfork,		"vfork"		}, /* 190 */
 	{ 2,	0,	sys_getrlimit,		"getrlimit"	}, /* 191 */
 	{ 6,	0,	sys_mmap,		"mmap2"		}, /* 192 */
 	{ 2,	TF,	sys_truncate64,		"truncate64"	}, /* 193 */
-	{ 2,	TF,	sys_ftruncate64,	"ftruncate64"	}, /* 194 */
+	{ 2,	0,	sys_ftruncate64,	"ftruncate64"	}, /* 194 */
 	{ 2,	TF,	sys_stat64,		"stat64"	}, /* 195 */
 	{ 2,	TF,	sys_lstat64,		"lstat64"	}, /* 196 */
-	{ 2,	TF,	sys_fstat64,		"fstat64"	}, /* 197 */
+	{ 2,	0,	sys_fstat64,		"fstat64"	}, /* 197 */
 	{ 3,	TF,	sys_chown,		"lchown"	}, /* 198 */
 	{ 0,	0,	sys_getuid,		"getuid"	}, /* 199 */
 	{ 0,	0,	sys_getgid,		"getgid"	}, /* 200 */
@@ -251,7 +251,7 @@
 	{ 3,	0,	sys_getdents64,		"getdents64"	}, /* 220 */
 	{ 3,	0,	sys_fcntl,		"fcntl64"	}, /* 221 */
 	{ 4,	0,	sys_readahead,		"readahead"	}, /* 222 */
-	{ 4,	TF,	sys_sendfile64,		"sendfile64"	}, /* 223 */
+	{ 4,	0,	sys_sendfile64,		"sendfile64"	}, /* 223 */
 	{ 5,	TF,	sys_setxattr,		"setxattr"	}, /* 224 */
 	{ 5,	TF,	sys_setxattr,		"lsetxattr"	}, /* 225 */
 	{ 5,	0,	sys_fsetxattr,		"fsetxattr"	}, /* 226 */
diff -upk.orig strace-4.5.12.orig/linux/s390x/syscallent.h strace-4.5.12/linux/s390x/syscallent.h
--- strace-4.5.12.orig/linux/s390x/syscallent.h	2005-05-09 07:54:30 +0000
+++ strace-4.5.12/linux/s390x/syscallent.h	2005-06-10 11:37:30 +0000
@@ -207,14 +207,14 @@
 	{ 4,	TS,	sys_rt_sigtimedwait,	"rt_sigtimedwait"}, /* 177 */
 	{ 3,	TS,	sys_rt_sigqueueinfo,	"rt_sigqueueinfo"}, /* 178 */
 	{ 2,	TS,	sys_rt_sigsuspend,	"rt_sigsuspend"	}, /* 179 */
-	{ 5,	TF,	sys_pread,		"pread"		}, /* 180 */
-	{ 5,	TF,	sys_pwrite,		"pwrite"	}, /* 181 */
+	{ 5,	0,	sys_pread,		"pread"		}, /* 180 */
+	{ 5,	0,	sys_pwrite,		"pwrite"	}, /* 181 */
 	{ -1,	0,	printargs,		"SYS_182"	}, /* 182 */
 	{ 2,	TF,	sys_getcwd,		"getcwd"	}, /* 183 */
 	{ 2,	0,	sys_capget,		"capget"	}, /* 184 */
 	{ 2,	0,	sys_capset,		"capset"	}, /* 185 */
 	{ 2,	TS,	sys_sigaltstack,	"sigaltstack"	}, /* 186 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"	}, /* 187 */
+	{ 4,	0,	sys_sendfile,		"sendfile"	}, /* 187 */
 	{ 5,	0,	sys_getpmsg,		"getpmsg"	}, /* 188 */
 	{ 5,	0,	sys_putpmsg,		"putpmsg"	}, /* 189 */
 	{ 0,	TP,	sys_vfork,		"vfork"		}, /* 190 */
@@ -250,7 +250,7 @@
 	{ 3,	0,	sys_getdents64,		"getdents64"	}, /* 220 */
 	{ -1,	0,	printargs,		"SYS_221"	}, /* 221 */
 	{ 4,	0,	sys_readahead,		"readahead"	}, /* 222 */
-	{ 4,	TF,	sys_sendfile64,		"sendfile64"	}, /* 223 */
+	{ 4,	0,	sys_sendfile64,		"sendfile64"	}, /* 223 */
 	{ 5,	TF,	sys_setxattr,		"setxattr"	}, /* 224 */
 	{ 5,	TF,	sys_setxattr,		"lsetxattr"	}, /* 225 */
 	{ 5,	0,	sys_fsetxattr,		"fsetxattr"	}, /* 226 */
diff -upk.orig strace-4.5.12.orig/linux/sh/syscallent.h strace-4.5.12/linux/sh/syscallent.h
--- strace-4.5.12.orig/linux/sh/syscallent.h	2003-11-13 22:32:28 +0000
+++ strace-4.5.12/linux/sh/syscallent.h	2005-06-08 08:55:32 +0000
@@ -211,14 +211,14 @@
 	{ 3,	TS,	sys_rt_sigqueueinfo,    "rt_sigqueueinfo"}, /* 178 */
 	{ 2,	TS,	sys_rt_sigsuspend,	"rt_sigsuspend"	}, /* 179 */
 
-	{ 6,	TF,	sys_pread,		"pread"		}, /* 180 */
-	{ 6,	TF,	sys_pwrite,		"pwrite"	}, /* 181 */
+	{ 6,	0,	sys_pread,		"pread"		}, /* 180 */
+	{ 6,	0,	sys_pwrite,		"pwrite"	}, /* 181 */
 	{ 3,	TF,	sys_chown,		"chown"		}, /* 182 */
 	{ 2,	0,	sys_getcwd,		"getcwd"	}, /* 183 */
 	{ 2,	0,	sys_capget,		"capget"	}, /* 184 */
 	{ 2,	0,	sys_capset,		"capset"	}, /* 185 */
 	{ 2,	TS,	sys_sigaltstack,	"sigaltstack"	}, /* 186 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"	}, /* 187 */
+	{ 4,	0,	sys_sendfile,		"sendfile"	}, /* 187 */
 	{ 5,	0,	printargs,		"SYS_188"	}, /* 188 */
 	{ 5,	0,	printargs,		"SYS_189"	}, /* 189 */
 	{ 0,	TP,	sys_vfork,		"vfork"		}, /* 190 */
@@ -228,7 +228,7 @@
 	{ 5,	0,	sys_ftruncate64,	"ftruncate64"	}, /* 194 */
 	{ 2,	TF,	sys_stat64,		"stat64"	}, /* 195 */
 	{ 2,	TF,	sys_lstat64,		"lstat64"	}, /* 196 */
-	{ 2,	TF,	sys_fstat64,		"fstat64"	}, /* 197 */
+	{ 2,	0,	sys_fstat64,		"fstat64"	}, /* 197 */
 /*TODO*/{ 3,	TF,	printargs,		"lchown32"	}, /* 198 */
 /*TODO*/{ 0,	0,	printargs,		"getuid32"	}, /* 199 */
 
diff -upk.orig strace-4.5.12.orig/linux/sh64/syscallent.h strace-4.5.12/linux/sh64/syscallent.h
--- strace-4.5.12.orig/linux/sh64/syscallent.h	2003-11-13 22:32:28 +0000
+++ strace-4.5.12/linux/sh64/syscallent.h	2005-06-08 08:55:32 +0000
@@ -208,24 +208,24 @@
 	{ 4,	TS,	sys_rt_sigtimedwait,	"rt_sigtimedwait"}, /* 177 */
 	{ 3,	TS,	sys_rt_sigqueueinfo,    "rt_sigqueueinfo"}, /* 178 */
 	{ 2,	TS,	sys_rt_sigsuspend,	"rt_sigsuspend"	}, /* 179 */
-	{ 5,	TF,	sys_pread,		"pread"		}, /* 180 */
-	{ 5,	TF,	sys_pwrite,		"pwrite"	}, /* 181 */
+	{ 5,	0,	sys_pread,		"pread"		}, /* 180 */
+	{ 5,	0,	sys_pwrite,		"pwrite"	}, /* 181 */
 	{ 3,	TF,	sys_chown,		"chown"		}, /* 182 */
 	{ 2,	TF,	sys_getcwd,		"getcwd"	}, /* 183 */
 	{ 2,	0,	sys_capget,		"capget"	}, /* 184 */
 	{ 2,	0,	sys_capset,		"capset"	}, /* 185 */
 	{ 2,	TS,	sys_sigaltstack,	"sigaltstack"	}, /* 186 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"	}, /* 187 */
+	{ 4,	0,	sys_sendfile,		"sendfile"	}, /* 187 */
 	{ 5,	0,	printargs,		"SYS_188"	}, /* 188 */
 	{ 5,	0,	printargs,		"SYS_189"	}, /* 189 */
 	{ 0,	TP,	sys_vfork,		"vfork"		}, /* 190 */
 	{ 2,	0,	printargs,		"getrlimit"	}, /* 191 */
 	{ 6,	0,	sys_mmap,		"mmap2"		}, /* 192 */
 	{ 2,	TF,	sys_truncate64,		"truncate64"	}, /* 193 */
-	{ 2,	TF,	sys_ftruncate64,	"ftruncate64"	}, /* 194 */
+	{ 2,	0,	sys_ftruncate64,	"ftruncate64"	}, /* 194 */
 	{ 2,	TF,	sys_stat64,		"stat64"	}, /* 195 */
 	{ 2,	TF,	sys_lstat64,		"lstat64"	}, /* 196 */
-	{ 2,	TF,	sys_fstat64,		"fstat64"	}, /* 197 */
+	{ 2,	0,	sys_fstat64,		"fstat64"	}, /* 197 */
 	{ 3,	TF,	sys_chown,		"lchown32"	}, /* 198 */
 	{ 0,	0,	sys_getuid,		"getuid32"	}, /* 199 */
 	{ 0,	0,	printargs,		"getgid32"	}, /* 200 */
diff -upk.orig strace-4.5.12.orig/linux/sparc/syscallent.h strace-4.5.12/linux/sparc/syscallent.h
--- strace-4.5.12.orig/linux/sparc/syscallent.h	2004-10-06 22:21:05 +0000
+++ strace-4.5.12/linux/sparc/syscallent.h	2005-06-08 08:55:32 +0000
@@ -37,7 +37,7 @@
 	{ 0,	0,	sys_sync,	"sync" },		/* 36 */
 	{ 2,	TS,	sys_kill,	"kill" },		/* 37 */
 	{ 2,	TF,	sys_stat,	"stat" },		/* 38 */
-	{ 4,	TF,	sys_sendfile,	"sendfile" },		/* 39 */
+	{ 4,	0,	sys_sendfile,	"sendfile" },		/* 39 */
 	{ 2,	TF,	sys_lstat,	"lstat" },		/* 40 */
 	{ 2,	0,	sys_dup,	"dup" },		/* 41 */
 	{ 0,	0,	sys_pipe,	"pipe" },		/* 42 */
@@ -61,12 +61,12 @@
 	{ 1,	0,	sys_umask,	"umask" },		/* 60 */
 	{ 1,	TF,	sys_chroot,	"chroot" },		/* 61 */
 	{ 2,	0,	sys_fstat,	"fstat" },		/* 62 */
-	{ 2,	TF,	sys_fstat64,	"fstat64" },		/* 63 */
+	{ 2,	0,	sys_fstat64,	"fstat64" },		/* 63 */
 	{ 0,	0,	sys_getpagesize,"getpagesize" },	/* 64 */
 	{ 3,	0,	sys_msync,	"msync" },		/* 65 */
 	{ 0,	TP,	sys_vfork,	"vfork" },		/* 66 */
-	{ 5,	TF,	sys_pread,	"pread" },		/* 67 */
-	{ 5,	TF,	sys_pwrite,	"pwrite" },		/* 68 */
+	{ 5,	0,	sys_pread,	"pread" },		/* 67 */
+	{ 5,	0,	sys_pwrite,	"pwrite" },		/* 68 */
 	{ 0,    0,	sys_geteuid,	"geteuid32" },		/* 69 */
 	{ 0,	0,	sys_getegid,	"getegid32" },		/* 70 */
 	{ 6,	0,	sys_mmap,	"mmap" },		/* 71 */
@@ -138,7 +138,7 @@
 	{ 1,	TF,	sys_rmdir,	"rmdir" },		/* 137 */
 	{ 2,	TF,	sys_utimes,	"utimes" },		/* 138 */
 	{ 2,	TF,	sys_stat64,	"stat64" },		/* 139 */
-	{ 4,    TF,	sys_sendfile64,	"sendfile64" },		/* 140 */
+	{ 4,    0,	sys_sendfile64,	"sendfile64" },		/* 140 */
 	{ 3,	TN,	sys_getpeername,"getpeername" },	/* 141 */
 	{ 5,    0,	sys_futex,	"futex" },		/* 142 */
 	{ 0,	0,	printargs,	"gettid" },		/* 143 */
diff -upk.orig strace-4.5.12.orig/linux/sparc64/syscallent.h strace-4.5.12/linux/sparc64/syscallent.h
--- strace-4.5.12.orig/linux/sparc64/syscallent.h	2004-10-06 22:20:53 +0000
+++ strace-4.5.12/linux/sparc64/syscallent.h	2005-06-08 08:55:32 +0000
@@ -37,7 +37,7 @@
 	{ 0,	0,	sys_sync,	"sync" },		/* 36 */
 	{ 2,	TS,	sys_kill,	"kill" },		/* 37 */
 	{ 2,	TF,	sys_stat,	"stat" },		/* 38 */
-	{ 4,	TF,	sys_sendfile,	"sendfile" },		/* 39 */
+	{ 4,	0,	sys_sendfile,	"sendfile" },		/* 39 */
 	{ 2,	TF,	sys_lstat,	"lstat" },		/* 40 */
 	{ 2,	0,	sys_dup,	"dup" },		/* 41 */
 	{ 0,	0,	sys_pipe,	"pipe" },		/* 42 */
@@ -61,12 +61,12 @@
 	{ 1,	0,	sys_umask,	"umask" },		/* 60 */
 	{ 1,	TF,	sys_chroot,	"chroot" },		/* 61 */
 	{ 2,	0,	sys_fstat,	"fstat" },		/* 62 */
-	{ 2,	TF,	sys_fstat64,	"fstat64" },		/* 63 */
+	{ 2,	0,	sys_fstat64,	"fstat64" },		/* 63 */
 	{ 0,	0,	sys_getpagesize,"getpagesize" },	/* 64 */
 	{ 3,	0,	sys_msync,	"msync" },		/* 65 */
 	{ 0,	TP,	sys_vfork,	"vfork" },		/* 66 */
-	{ 5,	TF,	sys_pread,	"pread" },		/* 67 */
-	{ 5,	TF,	sys_pwrite,	"pwrite" },		/* 68 */
+	{ 5,	0,	sys_pread,	"pread" },		/* 67 */
+	{ 5,	0,	sys_pwrite,	"pwrite" },		/* 68 */
 	{ 0,    0,	sys_geteuid,	"geteuid32" },		/* 69 */
 	{ 0,	0,	sys_getegid,	"getegid32" },		/* 70 */
 	{ 6,	0,	sys_mmap,	"mmap" },		/* 71 */
@@ -138,7 +138,7 @@
 	{ 1,	TF,	sys_rmdir,	"rmdir" },		/* 137 */
 	{ 2,	TF,	sys_utimes,	"utimes" },		/* 138 */
 	{ 2,	TF,	sys_stat64,	"stat64" },		/* 139 */
-	{ 4,    TF,	sys_sendfile64,	"sendfile64" },		/* 140 */
+	{ 4,    0,	sys_sendfile64,	"sendfile64" },		/* 140 */
 	{ 3,	TN,	sys_getpeername,"getpeername" },	/* 141 */
 	{ 5,    0,	sys_futex,	"futex" },		/* 142 */
 	{ 0,	0,	printargs,	"gettid" },		/* 143 */
diff -upk.orig strace-4.5.12.orig/linux/sparc64/syscallent2.h strace-4.5.12/linux/sparc64/syscallent2.h
--- strace-4.5.12.orig/linux/sparc64/syscallent2.h	2004-10-06 22:20:37 +0000
+++ strace-4.5.12/linux/sparc64/syscallent2.h	2005-06-08 08:55:32 +0000
@@ -37,7 +37,7 @@
 	{ 0,	0,	sys_sync,	"sync" },		/* 36 */
 	{ 2,	TS,	sys_kill,	"kill" },		/* 37 */
 	{ 2,	TF,	sys_stat,	"stat" },		/* 38 */
-	{ 4,	TF,	sys_sendfile,	"sendfile" },		/* 39 */
+	{ 4,	0,	sys_sendfile,	"sendfile" },		/* 39 */
 	{ 2,	TF,	sys_lstat,	"lstat" },		/* 40 */
 	{ 2,	0,	sys_dup,	"dup" },		/* 41 */
 	{ 0,	0,	sys_pipe,	"pipe" },		/* 42 */
@@ -61,12 +61,12 @@
 	{ 1,	0,	sys_umask,	"umask" },		/* 60 */
 	{ 1,	TF,	sys_chroot,	"chroot" },		/* 61 */
 	{ 2,	0,	sys_fstat,	"fstat" },		/* 62 */
-	{ 2,	TF,	sys_fstat64,	"fstat64" },		/* 63 */
+	{ 2,	0,	sys_fstat64,	"fstat64" },		/* 63 */
 	{ 0,	0,	sys_getpagesize,"getpagesize" },	/* 64 */
 	{ 3,	0,	sys_msync,	"msync" },		/* 65 */
 	{ 0,	TP,	sys_vfork,	"vfork" },		/* 66 */
-	{ 5,	TF,	sys_pread,	"pread" },		/* 67 */
-	{ 5,	TF,	sys_pwrite,	"pwrite" },		/* 68 */
+	{ 5,	0,	sys_pread,	"pread" },		/* 67 */
+	{ 5,	0,	sys_pwrite,	"pwrite" },		/* 68 */
 	{ 0,    0,	sys_geteuid,	"geteuid32" },		/* 69 */
 	{ 0,	0,	sys_getegid,	"getegid32" },		/* 70 */
 	{ 6,	0,	sys_mmap,	"mmap" },		/* 71 */
@@ -138,7 +138,7 @@
 	{ 1,	TF,	sys_rmdir,	"rmdir" },		/* 137 */
 	{ 2,	TF,	sys_utimes,	"utimes" },		/* 138 */
 	{ 2,	TF,	sys_stat64,	"stat64" },		/* 139 */
-	{ 4,    TF,	sys_sendfile64,	"sendfile64" },		/* 140 */
+	{ 4,    0,	sys_sendfile64,	"sendfile64" },		/* 140 */
 	{ 3,	TN,	sys_getpeername,"getpeername" },	/* 141 */
 	{ 5,    0,	sys_futex,	"futex" },		/* 142 */
 	{ 0,	0,	printargs,	"gettid" },		/* 143 */
diff -upk.orig strace-4.5.12.orig/linux/syscallent.h strace-4.5.12/linux/syscallent.h
--- strace-4.5.12.orig/linux/syscallent.h	2005-06-07 23:21:24 +0000
+++ strace-4.5.12/linux/syscallent.h	2005-06-10 11:37:30 +0000
@@ -221,8 +221,8 @@
 	{ 3,	TS,	sys_rt_sigqueueinfo,    "rt_sigqueueinfo"}, /* 178 */
 	{ 2,	TS,	sys_rt_sigsuspend,	"rt_sigsuspend"	}, /* 179 */
 
-	{ 5,	TF,	sys_pread,		"pread64", SYS_read }, /* 180 */
-	{ 5,	TF,	sys_pwrite,		"pwrite64", SYS_write }, /* 181 */
+	{ 5,	0,	sys_pread,		"pread64", SYS_read }, /* 180 */
+	{ 5,	0,	sys_pwrite,		"pwrite64", SYS_write }, /* 181 */
 #ifdef M68K
 	{ 3,	TF,	sys_chown,		"lchown"	}, /* 182 */
 #else
@@ -232,17 +232,17 @@
 	{ 2,	0,	sys_capget,		"capget"	}, /* 184 */
 	{ 2,	0,	sys_capset,		"capset"	}, /* 185 */
 	{ 2,	TS,	sys_sigaltstack,	"sigaltstack"	}, /* 186 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"	}, /* 187 */
+	{ 4,	0,	sys_sendfile,		"sendfile"	}, /* 187 */
 	{ 5,	0,	sys_getpmsg,		"getpmsg"	}, /* 188 */
 	{ 5,	0,	sys_putpmsg,		"putpmsg"	}, /* 189 */
 	{ 0,	TP,	sys_vfork,		"vfork", SYS_vfork }, /* 190 */
 	{ 2,	0,	sys_getrlimit,		"getrlimit"	}, /* 191 */
 	{ 6,	0,	sys_mmap,		"mmap2"		}, /* 192 */
 	{ 3,	TF,	sys_truncate64,		"truncate64"	}, /* 193 */
-	{ 3,	TF,	sys_ftruncate64,	"ftruncate64"	}, /* 194 */
+	{ 3,	0,	sys_ftruncate64,	"ftruncate64"	}, /* 194 */
 	{ 2,	TF,	sys_stat64,		"stat64"	}, /* 195 */
 	{ 2,	TF,	sys_lstat64,		"lstat64"	}, /* 196 */
-	{ 2,	TF,	sys_fstat64,		"fstat64"	}, /* 197 */
+	{ 2,	0,	sys_fstat64,		"fstat64"	}, /* 197 */
 	{ 3,	TF,	sys_chown,		"lchown32"	}, /* 198 */
 	{ 0,	0,	sys_getuid,		"getuid32"	}, /* 199 */
 
@@ -285,7 +285,7 @@
 	{ 2,	TF,	sys_removexattr,	"lremovexattr"	}, /* 236 */
 	{ 2,	0,	sys_fremovexattr,	"fremovexattr"	}, /* 237 */
 	{ 2,	TS,	sys_kill,		"tkill"		}, /* 238 */
-	{ 4,	TF,	sys_sendfile64,		"sendfile64"	}, /* 239 */
+	{ 4,	0,	sys_sendfile64,		"sendfile64"	}, /* 239 */
 	{ 5,	0,	sys_futex,		"futex"		}, /* 240 */
 	{ 3,	0,	sys_sched_setaffinity,	"sched_setaffinity" },/* 241 */
 	{ 3,	0,	sys_sched_getaffinity,	"sched_getaffinity" },/* 242 */
@@ -296,7 +296,7 @@
 	{ 5,	0,	sys_io_getevents,	"io_getevents"	}, /* 247 */
 	{ 3,	0,	sys_io_submit,		"io_submit"	}, /* 248 */
 	{ 3,	0,	sys_io_cancel,		"io_cancel"	}, /* 249 */
-	{ 5,	TF,	sys_fadvise64,		"fadvise64"	}, /* 250 */
+	{ 5,	0,	sys_fadvise64,		"fadvise64"	}, /* 250 */
 	{ 5,	0,	printargs,		"SYS_251"	}, /* 251 */
 	{ 1,	TP,	sys_exit,		"exit_group", __NR_exit_group }, /* 252 */
 	{ 4,	0,	printargs,		"lookup_dcookie"}, /* 253 */
@@ -315,10 +315,10 @@
 	{ 2,	0,	sys_clock_getres,	"clock_getres"	}, /* 266 */
 	{ 4,	0,	sys_clock_nanosleep,	"clock_nanosleep"}, /* 267 */
 	{ 3,	TF,	sys_statfs64,		"statfs64"	}, /* 268 */
-	{ 3,	TF,	sys_fstatfs64,		"fstatfs64"	}, /* 269 */
+	{ 3,	0,	sys_fstatfs64,		"fstatfs64"	}, /* 269 */
 	{ 3,	TS,	sys_tgkill,		"tgkill"	}, /* 270 */
 	{ 2,	TF,	sys_utimes,		"utimes"	}, /* 271 */
-	{ 6,	TF,	sys_fadvise64_64,	"fadvise64_64"	}, /* 272 */
+	{ 6,	0,	sys_fadvise64_64,	"fadvise64_64"	}, /* 272 */
 	{ 5,	0,	printargs,		"vserver"	}, /* 273 */
 	{ 6,	0,	sys_mbind,		"mbind"		}, /* 274 */
 	{ 5,	0,	sys_get_mempolicy,	"get_mempolicy"	}, /* 275 */
diff -upk.orig strace-4.5.12.orig/linux/x86_64/syscallent.h strace-4.5.12/linux/x86_64/syscallent.h
--- strace-4.5.12.orig/linux/x86_64/syscallent.h	2005-05-09 07:54:29 +0000
+++ strace-4.5.12/linux/x86_64/syscallent.h	2005-06-10 11:37:30 +0000
@@ -15,8 +15,8 @@
 	{ 4,	TS,	sys_rt_sigprocmask,	"rt_sigprocmask"},  /* 14 */
 	{ 1,	TS,	printargs,		"rt_sigreturn"	},  /* 15 */
 	{ 3,	0,	sys_ioctl,		"ioctl"		},  /* 16 */
-	{ 5,	TF,	sys_pread,		"pread"		},  /* 17 */
-	{ 5,	TF,	sys_pwrite,		"pwrite"	},  /* 18 */
+	{ 5,	0,	sys_pread,		"pread"		},  /* 17 */
+	{ 5,	0,	sys_pwrite,		"pwrite"	},  /* 18 */
 	{ 3,	0,	sys_readv,		"readv"		},  /* 19 */
 	{ 3,	0,	sys_writev,		"writev"	},  /* 20 */
 	{ 2,	TF,	sys_access,		"access"	},  /* 21 */
@@ -38,7 +38,7 @@
 	{ 1,	0,	sys_alarm,		"alarm"		},  /* 37 */
 	{ 3,	0,	sys_setitimer,		"setitimer"	},  /* 38 */
 	{ 0,	0,	sys_getpid,		"getpid"	},  /* 39 */
-	{ 4,	TF,	sys_sendfile,		"sendfile"	},  /* 40 */
+	{ 4,	0,	sys_sendfile,		"sendfile"	},  /* 40 */
 	{ 3,	TN,	sys_socket,		"socket"	},  /* 41 */
 	{ 3,	TN,	sys_connect,		"connect"	},  /* 42 */
 	{ 3,	TN,	sys_accept,		"accept"	},  /* 43 */
@@ -219,7 +219,7 @@
 	{ 1,	0,	printargs,		"set_tid_address"}, /* 218 */
 	{ 5,	0,	printargs,		"restart_syscall"}, /* 219 */
 	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 220 */
-	{ 4,	TF,	sys_fadvise64_64,	"fadvise64"	}, /* 221 */
+	{ 4,	0,	sys_fadvise64_64,	"fadvise64"	}, /* 221 */
 	{ 3,	0,	sys_timer_create,	"timer_create"	}, /* 222 */
 	{ 4,	0,	sys_timer_settime,	"timer_settime"	}, /* 223 */
 	{ 2,	0,	sys_timer_gettime,	"timer_gettime"	}, /* 224 */
diff -upk.orig strace-4.5.12.orig/svr4/syscallent.h strace-4.5.12/svr4/syscallent.h
--- strace-4.5.12.orig/svr4/syscallent.h	2001-03-27 13:57:48 +0000
+++ strace-4.5.12/svr4/syscallent.h	2005-06-08 08:55:32 +0000
@@ -414,10 +414,10 @@
 	{ -1,	0,	sys_unblock,		"unblock"	}, /* 186 */
 	{ -1,	0,	sys_cancelblock,	"cancelblock"	}, /* 187 */
 	{ -1,	0, 	printargs,		"SYS_188"	}, /* 188 */
-	{ -1,	TF,	sys_pread,		"pread"		}, /* 189 */
-	{ -1,	TF,	sys_pwrite,		"pwrite"	}, /* 190 */
+	{ -1,	0,	sys_pread,		"pread"		}, /* 189 */
+	{ -1,	0,	sys_pwrite,		"pwrite"	}, /* 190 */
 	{ -1,	TF,	sys_truncate,		"truncate"	}, /* 191 */
-	{ -1,	TF,	sys_ftruncate,		"ftruncate"	}, /* 192 */
+	{ -1,	0,	sys_ftruncate,		"ftruncate"	}, /* 192 */
 	{ -1,	0,	sys_lwpkill,		"lwpkill"	}, /* 193 */
 	{ -1,	0,	sys_sigwait,		"sigwait"	}, /* 194 */
 	{ -1,	0,	sys_fork1,		"fork1"		}, /* 195 */
@@ -442,9 +442,9 @@
 	{ -1,	0,	printargs,		"SYS_214"	}, /* 214 */
 	{ -1,	0,	printargs,		"SYS_215"	}, /* 215 */
 #if UNIXWARE >= 7
-	{ -1,	TF,	sys_fstatvfs64,		"fstatvfs64"	}, /* 216 */
+	{ -1,	0,	sys_fstatvfs64,		"fstatvfs64"	}, /* 216 */
 	{ -1,	TF,	sys_statvfs64,		"statvfs64"	}, /* 217 */
-	{ -1,	TF,	sys_ftruncate64,	"ftruncate64"	}, /* 218 */
+	{ -1,	0,	sys_ftruncate64,	"ftruncate64"	}, /* 218 */
 	{ -1,	TF,	sys_truncate64,		"truncate64"	}, /* 219 */
 	{ -1,	0,	sys_getrlimit64,	"getrlimit64"	}, /* 220 */
 	{ -1,	0,	sys_setrlimit64,	"setrlimit64"	}, /* 221 */
@@ -550,8 +550,8 @@
 	{ -1,	0,	sys_lwp_cond_wait,	"lwp_cond_wait"}, /* 170 */
 	{ -1,	0,	sys_lwp_cond_signal,	"lwp_cond_signal"}, /* 171 */
 	{ -1,	0,	sys_lwp_cond_broadcast,	"lwp_cond_broadcast"}, /* 172 */
-	{ -1,	TF,	sys_pread,		"pread"		}, /* 173 */
-	{ -1,	TF,	sys_pwrite,		"pwrite"	}, /* 174 */
+	{ -1,	0,	sys_pread,		"pread"		}, /* 173 */
+	{ -1,	0,	sys_pwrite,		"pwrite"	}, /* 174 */
 	{ -1,	0,	sys_llseek,		"llseek"	}, /* 175 */
 	{ -1,	0,	sys_inst_sync,		"inst_sync"	}, /* 176 */
 	{ -1,	0,	printargs,		"srmlimitsys"	}, /* 177 */
